wee_slack.py

```diff
@@ -3126,24 +3140,38 @@ def command_register(data, current_buffer, args):
         "https://slack.com/api/oauth.access?"
         "client_id={}&client_secret={}&code={}"
     ).format(CLIENT_ID, CLIENT_SECRET, oauth_code)
-    ret = urllib.urlopen(uri).read()
-    d = json.loads(ret)
-    if not d["ok"]:
-        w.prnt("",
-            "ERROR: Couldn't get Slack OAuth token: {}".format(d['error']))
-        return
+    params = {'useragent': 'wee_slack {}'.format(SCRIPT_VERSION)}
+    w.hook_process_hashtable('url:', params, config.slack_timeout, "", "")
+    w.hook_process_hashtable("url:{}".format(uri), params, config.slack_timeout, "command_register_callback", "")
+    return
 
-    if config.is_default('slack_api_token'):
-        w.config_set_plugin('slack_api_token', d['access_token'])
-    else:
-        # Add new token to existing set, joined by comma.
-        tok = config.get_string('slack_api_token')
-        w.config_set_plugin('slack_api_token',
-                            ','.join([tok, d['access_token']]))
+@utf8_decode
+def command_register_callback(data, command, return_code, out, err):
+    if return_code == 0:
+        if len(out) > 0:
+            d = json.loads(out)
```

Comment:
To make this more readable and prevent deep nesting, it's better to write this as

`if return code != 0: ` and then do the error handling + return. It also means matching up the condition and the code that runs is easier.